packages/x-data-grid/src/components/GridRow.tsx

```diff
@@ -64,7 +64,7 @@ export interface GridRowProps extends React.HTMLAttributes<HTMLDivElement> {
   onDoubleClick?: React.MouseEventHandler<HTMLDivElement>;
   onMouseEnter?: React.MouseEventHandler<HTMLDivElement>;
   onMouseLeave?: React.MouseEventHandler<HTMLDivElement>;
-  [x: string]: any; // Allow custom attributes like data-* and aria-*
+  [x: `data-${string}`]: string;
```

Comment:
@mui/xgrid Are you sure we need to do this?
What problem is it solving? 🤔 
No other X package is doing it, and neither `Core` is doing it. 🤔

Reply:
shouldn't these custom attributes (from the comment in the code) be covered with `extends React.HTMLAttributes<HTMLDivElement>`?

Reply:
For context—no other package is providing this explicit type, if anyone needs to provide it, they are probably fighting TS as well.
Such props work on HTML and JSX elements as any prop with a `-` is ignored by TS.
Given that in our case these props are passed through `slotProps`, this causes slightly more problems, because they will show an error. 🤔